src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -1390,7 +1399,14 @@ fsm_follow_new_primary(Keeper *keeper)
 	}
 
 	/* now, in case we have an init state file around, remove it */
-	return unlink_file(config->pathnames.init);
+	if (!unlink_file(config->pathnames.init))
+	{
+		/* errors have already been logged */
+		return false;
+	}
+
+	/* finally, check that we're on the same timeline as the new primary */
+	return standby_check_timeline_with_upstream(postgres);
```

Comment:
Do we really need this here? This function is used for the transition from `secondary` to `catchingup`. I think it's enough to only do it in the transition from `catchingup` back to `secondary`.

Reply:
Good point. Because we use the same transition function in both places, and there's a lot of common code, I just added a check that way:

``` c
	/*
	 * Finally, check that we're on the same timeline as the new primary when
	 * assigned secondary as a goal state. This transition function is also
	 * used when going from secondary to catchingup, as the primary might have
	 * changed also in that situation.
	 */
	if (keeper->state.assigned_role == SECONDARY_STATE)
	{
		return standby_check_timeline_with_upstream(postgres);
	}
```